### DISCUSSION TITLE
Coral thinks my tree is a person

### DISCUSSION THREAD

**Discussion by ronisaacson** [blakeblackshear/frigate#6147] [Ask A Question] [answered]
There's one tree in my yard that the Coral has decided is a person. It's far away from the camera and I guess it does look sort of person-like when the light hits it just right?

I was following the other thread about stationary objects but I'm not sure how to apply that here. I don't need to do global tuning because the person detection is working really well otherwise. I could probably use a very small mask to block the face-like part, but this would also interfere with my driveway.

Is there a targeted way to say "this is not a person" based on a specific image or coordinates?

**Comment by NickM-27** [comment, marked as answer]
object masks are based on the bottom center of the bounding box that is put around an object. In the case of a tree, it is a good candidate for a small object mask that goes at the bottom of the tree

**Reply by ronisaacson** [reply]
Thanks, I watched debug view until it popped up and and found the bounding box. I didn't know you could define object-specific masks with that granularity but now I found it in the config ref.

Are the bounding boxes for detected objects saved with the event clips for later review? Or are they only available in the live debug view?

Also a related question: is it possible to clear a previously-detected object? There was a shadow on a different camera that the Coral thought was a bear, and to be fair, it really did look like a bear. I'd like to tell it "hey, there was never a bear here, so please clear the thumbnail and go back to showing a black box in the object list."

PS: in case you're curious, the person/tree, I don't really see it myself but hey:

![image](https://user-images.githubusercontent.com/13400038/232903860-c815d87f-5f31-4475-b757-dee8319bb688.png)

**Reply by NickM-27** [reply]
> Are the bounding boxes for detected objects saved with the event clips for later review? Or are they only available in the live debug view?

currently, no they are not. This is something we are looking to change in an upcoming release.

> Also a related question: is it possible to clear a previously-detected object? There was a shadow on a different camera that the Coral thought was a bear, and to be fair, it really did look like a bear. I'd like to tell it "hey, there was never a bear here, so please clear the thumbnail and go back to showing a black box in the object list."

I mean you can delete the event if that is what you mean.

> PS: in case you're curious, the person/tree, I don't really see it myself but hey:

the current model used in frigate is a Google demo model trained on the COCO data set [which looks nothing like a security camera perspective](https://imgur.com/a/CNImofJ) so that would explain it. [Frigate+ ](https://docs.frigate.video/integrations/plus) will provide better models in the future.

**Reply by ronisaacson** [reply]
Also for anyone else who finds this thread in the future, since actually the docs don't make it very clear how to do this:

```
cameras:
  driveway:
    ffmpeg:
      ...
    objects:
      filters:
        person:
          mask:
            - 535,300,545,300,545,285,535,285
```

This will add an object mask at the listed position for only this object, on only this camera. All of the global object settings are still inherited. The example in the docs is an object mask at the same position on all cameras, which is probably not what you want in most cases.

**Reply by ronisaacson** [reply]
> > Also a related question: is it possible to clear a previously-detected object? There was a shadow on a different camera that the Coral thought was a bear, and to be fair, it really did look like a bear. I'd like to tell it "hey, there was never a bear here, so please clear the thumbnail and go back to showing a black box in the object list."
> 
> I mean you can delete the event if that is what you mean.

Not quite... if you click on a camera, you'll see the live view for that camera, and then underneath it says "tracked objects" and a list of all enabled object types. Some of them are black boxes because it's never detected one of those. Example:

![image](https://user-images.githubusercontent.com/13400038/232912474-180b2227-a66d-4f12-a797-97e49e6ccebf.png)

But the BEAR one has a thumbnail there for something it thought was a bear. I deleted the corresponding event but the not-a-bear thumbnail still shows up in the tracked objects list. Is there a way to clear this and reset it to an empty black box?

**Reply by NickM-27** [reply]
those images come directly from the events. However, they are cached. It will go away when the cached image resets in 5 minutes I think (don't remember exactly what it is)

**Reply by ronisaacson** [reply]
Thanks, I refreshed after waiting a bit and it did reset. I appreciate the help!

**Comment by FrigateEnthusiast** [comment]
I had the same thing happen. Two small sapling like trees that crossed each other. I guess it detected that as legs of a person. I just bumped my confidence score up from 60 to 63.

objects:
    track:
        - person
        - car
    filters:
        person:
           min_area: 3000
           max_area: 8000000
           threshold: 0.63
